File: app/src/main/java/com/ekoapp/sdk/SampleApp.java
package com.ekoapp.sdk;

import androidx.multidex.MultiDexApplication;

import com.ekoapp.ekosdk.EkoClient;
import com.ekoapp.sdk.common.preferences.SamplePreferences;
import com.ekoapp.sdk.publisher.EkoStreamPublisherClient;

public class SampleApp extends MultiDexApplication {

    private static SampleApp APP;


    @Override
    public void onCreate() {
        super.onCreate();
        APP = this;
        EkoClient.setup(SamplePreferences.getApiKey().get());
        EkoStreamPublisherClient.setup(EkoClient.getConfiguration());
    }

    public static SampleApp get() {
        return APP;
    }
}

File: app/src/main/java/com/ekoapp/simplechat/SampleApp.kt
package com.ekoapp.simplechat

import androidx.multidex.MultiDexApplication
import com.ekoapp.ekosdk.EkoClient
import com.ekoapp.push.EkoBaidu

class SampleApp : MultiDexApplication() {

    companion object {
        private lateinit var APP: SampleApp

        fun get(): SampleApp {
            return APP
        }
    }

    override fun onCreate() {
        super.onCreate()
        APP = this

        EkoClient.setup(SimplePreferences.getApiKey().get())
                .andThen(EkoBaidu.create(this).setup("BZ2CnTh6qphSUl66c16Xk7AG"))
                .subscribe()
    }
}